### PR TITLE
fixes #12830 - Support rebooting hosts when setting multiple hosts to build

### PR DESCRIPTION
**Pull request by timogoebel**
When setting multiple hosts to build, this commit adds support to reboot the hosts in addition to enabling rebuild.

### CODE REVIEW

File: test/functional/hosts_controller_test.rb [hunk 1]
```diff
@@ -505,11 +505,22 @@ def set_remote_user_to(user)
...
       assert !@host2.build
       multiple_hosts_submit_request('build', [@host1.id, @host2.id],
-                                    'The selected hosts will execute a build operation on next reboot')
+                                    'The selected hosts will execute a build operation on next reboot',
+                                    {:host => { :build => 0 }})
+      assert Host.find(@host1).build
+      assert Host.find(@host2).build
+    end
+
+    test 'build with reboot' do
```
Comment (domcleal) [inline test/functional/hosts_controller_test.rb:518]: This test needs to check that `host.power.reboot` was called.  I think it's probably returning false for `supports_power_and_running` in the test so is skipping the step, so I'd suggest adding a `Host::Managed.any_instance.expects(:supports_power_and_running).twice.returns(true)` or similar, then expect the `.reboot` call too.

File: app/models/host/managed.rb [hunk 2]
```diff
@@ -786,6 +786,14 @@ def power
     end
   end
 
+  def supports_power_and_running?
```
Comment (domcleal) [inline app/models/host/managed.rb:789]: Could you add a unit test to host_test.rb for this please?

File: test/functional/hosts_controller_test.rb (hunk 1 above)
Comment (timogoebel) [inline test/functional/hosts_controller_test.rb:518]: It does now. Hope, that's ok now. :-)

File: app/models/host/managed.rb (hunk 2 above)
Comment (timogoebel) [inline app/models/host/managed.rb:789]: @domcleal : I added two tests to test/unit/host_test.rb. What do you think of them?

**Comment by domcleal** [comment]
[test] to check for failures when rebased.

**Comment by timogoebel** [comment]
@domcleal : Tests are only unstable for Katello (which we've had before). I think, we're ready to go. :-)

**Comment by domcleal** [comment]
Merged as d6814ff, thanks @timogoebel!

**Pull request closed without merging**